src/Constructorio_NET.Tests/client/modules/SearchTests.cs

```diff
@@ -128,6 +128,40 @@ public async Task GetSearchResultsShouldReturnResultWithRefinedContent()
             Assert.IsNotNull(res.ResultId, "ResultId should exist");
         }
 
+        [Test]
+        public async Task GetSearchResultsShouldReturnResultWithHiddenFields()
+        {
+            string requestedHiddenField = "testField";
+            SearchRequest req = new SearchRequest("item1")
+            {
+                UserInfo = this.UserInfo,
+                HiddenFields = new List<string> { requestedHiddenField }
+            };
+            ConstructorIO constructorio = new ConstructorIO(this.Config);
+            SearchResponse res = await constructorio.Search.GetSearchResults(req);
+            var returnedHiddenfield = res.Response.Results[0].Data.Metadata[requestedHiddenField];
+
+            Assert.NotNull(res.ResultId, "Result id exists");
+            Assert.NotNull(returnedHiddenfield, "Hidden field returned");
+        }
+
+        [Test]
+        public async Task GetSearchResultsShouldReturnResultWithHiddenFacets()
+        {
+            string requestedHiddenFacet = "Brand";
+            SearchRequest req = new SearchRequest("item1")
+            {
+                UserInfo = this.UserInfo,
+                HiddenFacets = new List<string> { requestedHiddenFacet }
+            };
+            ConstructorIO constructorio = new ConstructorIO(this.Config);
+            SearchResponse res = await constructorio.Search.GetSearchResults(req);
+            string returnedHiddenFacet = res.Response.Facets.Find(el => el.Hidden).DisplayName;
+
+            Assert.NotNull(res.ResultId, "Result id exists");
+            Assert.True(requestedHiddenFacet == returnedHiddenFacet, "Hidden facet returned");
```

Comment:
Should we add a check here to see if the hidden attribute on the facet is set to true?